src/ucp/rma/put_offload.c

```diff
@@ -14,8 +14,68 @@
 #include <ucp/core/ucp_request.inl>
 #include <ucp/dt/datatype_iter.inl>
 #include <ucp/proto/proto_multi.inl>
+#include <ucp/proto/proto_single.inl>
 
 
+static ucs_status_t ucp_proto_put_offload_short_progress(uct_pending_req_t *self)
+{
+    ucp_request_t                   *req = ucs_container_of(self, ucp_request_t,
+                                                            send.uct);
+    const ucp_proto_single_priv_t *spriv = req->send.proto_config->priv;
+    ucp_ep_t                         *ep = req->send.ep;
+    ucs_status_t status;
+    uct_rkey_t tl_rkey;
+
+    tl_rkey = ucp_rma_request_get_tl_rkey(req, spriv->super.rkey_index);
+    status  = uct_ep_put_short(ep->uct_eps[spriv->super.lane],
+                               req->send.dt_iter.type.contig.buffer,
+                               req->send.dt_iter.length,
+                               req->send.rma.remote_addr, tl_rkey);
+    if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
+        req->send.lane = spriv->super.lane; /* for pending add */
+        return status;
+    }
+
+    /* UCS_INPROGRESS is not expected */
+    ucs_assert((status == UCS_OK) || UCS_STATUS_IS_ERR(status));
+
+    ucp_datatype_iter_cleanup(&req->send.dt_iter, UCS_BIT(UCP_DATATYPE_CONTIG));
+    ucp_request_complete_send(req, status);
+    return UCS_OK;
+}
+
+static ucs_status_t
+ucp_proto_put_offload_short_init(const ucp_proto_init_params_t *init_params)
+{
+    ucp_proto_single_init_params_t params = {
+        .super.super         = *init_params,
+        .super.latency       = -150e-9,
+        .super.overhead      = 0,
+        .super.cfg_thresh    = UCS_MEMUNITS_AUTO,
+        .super.cfg_priority  = 0,
+        .super.min_frag_offs = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.max_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.put.max_short),
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RECV_ZCOPY |
+                               UCP_PROTO_COMMON_INIT_FLAG_REMOTE_ACCESS,
+        .lane_type           = UCP_LANE_TYPE_RMA,
+        .tl_cap_flags        = UCT_IFACE_FLAG_PUT_SHORT,
+        .super.hdr_size      = 0
+    };
+
+    UCP_RMA_PROTO_INIT_CHECK(init_params, UCP_OP_ID_PUT);
+
+    return ucp_proto_single_init(&params);
+}
+
+static ucp_proto_t ucp_put_offload_short_proto = {
+    .name       = "put/short",
```

Comment:
`"put/offload/short"` to align with '"put/offload.zcopy"' below

Reply:
should we also fix the "get/<proto_type>" -> "get/oflload/<proto_type>" protocol names introduced in the #5807 PR?